### ISSUE TITLE
Styling of column and row headers not reflected when saving to excel.  

### ISSUE THREAD

**Issue by saihari** [pandas-dev/pandas#25185]
Following is the code I used to style the headers of both rows and columns and then saving the stylized data frame to an excel file.

```python
df = pd.DataFrame([1000000.0, 1000000.0], index=pd.bdate_range('2017-08-01', '2017-08-02'))

df.index = df.index.strftime("%Y-%m-%d")
styler = df.style
styler.set_table_styles(
# select the table header with th and set it right align
    [dict(selector="th", props=[("text-align", "right"),('color','green')])]   
)
styler.to_excel('styled.xlsx', engine='openpyxl')

```
#### Problem description

I am trying to stylize the headers of a data frame and then save that data frame to an excel file, but I am unable to see the stylized effects inside the excel file. 

#### Output of Jupyter Notebook 
![image](https://user-images.githubusercontent.com/9162599/52338222-cc3ead80-2a2f-11e9-92da-e051988693ca.png)

#### Output in excel file 
![image](https://user-images.githubusercontent.com/9162599/52338330-1e7fce80-2a30-11e9-9fbb-162bddd82c4a.png)

#### Expected Output
![image](https://user-images.githubusercontent.com/9162599/52339032-fe510f00-2a31-11e9-8826-3a3d5df214f7.png)

Pandas Version: '0.24.1'

**Comment by gfyoung** [comment]
I was unaware that our Excel IO functionality could even support that level of formatting (or styling) for `DataFrame`.  That being said, it would be nice if we could...

**Comment by WillAyd** [comment]
This is somewhat related to #22773

So right now there's not a great API to do this. This is controlled internally by the `header_style` property of the ExcelFormatter class:

https://github.com/pandas-dev/pandas/blob/683c7b55f5195fdf4f524239066cbf6f1301f0e7/pandas/io/formats/excel.py#L385

That's not part of the exposed API, nor is it an exposed attribute of the Styler object as that instantiates it locally:

https://github.com/pandas-dev/pandas/blob/683c7b55f5195fdf4f524239066cbf6f1301f0e7/pandas/io/formats/style.py#L179

So one option is to figure out a way to logically expose that, though I'm not sure if that would really jive with the HTML elements that you are styling above.

In any case investigation and PRs would certainly be welcome

**Comment by attack68** [comment]
closed as recorded in master tracker #42276
